crates/spirv-std/src/arch.rs

```diff
@@ -27,3 +27,47 @@ pub unsafe fn vector_extract_dynamic<T: Scalar, V: Vector<T>>(vector: V, index:
 
     result
 }
+
+/// Make a copy of a vector, with a single, variably selected,
+/// component modified.
+///
+/// # Safety
+/// Behavior is undefined if `index`’s value is greater than or equal to the
+/// number of components in `vector`.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpVectorInsertDynamic")]
+#[inline]
+pub unsafe fn vector_insert_dynamic<T: Scalar, V: Vector<T>>(vector: V, index: usize, element: T) -> V {
+    let mut result = V::default();
+
+    asm! {
+        "%vector = OpLoad _ {vector}",
+        "%new_vector = OpVectorInsertDynamic _ %vector {element} {index}",
+        "OpStore {result} %new_vector",
+        vector = in(reg) &vector,
+        index = in(reg) index,
+        element = in(reg) &element,
+        result = in(reg) &mut result,
+    }
+
+    result
+}
+
+/// Make a copy of `object`.
+/// 
+/// # Safety
+/// `T` must be a composite type (vector, array, or matrix).
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpCopyObject")]
+#[inline]
+pub unsafe fn copy_object<T>(object: T) -> T {
```

Comment:
I believe `OpCopyObject` is an extremely esoteric bizarre instruction only used in either compilers that have very strange hacks in how they treat SSA form (and is functionally useless), and mmmaybe strange decoration edge cases, and there's no use in exposing it as a rust function. However, there's no _harm_ in exposing it I guess.

Reply:
Removed it for now.